packages/core/src/forms/control/control.element.ts

```diff
@@ -338,4 +339,10 @@ export class CdsControl extends LitElement {
       this.hiddenLabel = true;
     }
   }
+
+  private assignSlotIfInControlGroup() {
+    if (this.parentElement?.hasAttribute('cds-control-group')) {
+      this.setAttribute('slot', 'controls');
+    }
+  }
```

Comment:
Prior the group assigned the slot name and would only do it once  when the control group was created, missing any controls that may have been added later.

The fix is, instead of the control group assigning the slot the control now will assign itself. This ensures if the control is dynamically created it can set the right slot name.